tools/test-my-pr.py

```diff
@@ -0,0 +1,116 @@
+
+
+import donate_cpu_lib as lib
+import argparse
+import os
+import sys
+import random
+import time
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Run this script from your branch with Cppcheck patch to check to verify your patch against current master. It will compare output of testing bunch of opensource packages')
+    parser.add_argument('-j', default=1, type=int, help='Concurency execution threads')
+    parser.add_argument('-p', default=1000, type=int, help='Count of packages to check')
+    parser.add_argument('-o', default='my_check_diff.log', help='Filename of result inside a working path dir')
+    parser.add_argument('--work-path', '--work-path=', default=lib.work_path, type=str, help='Working directory for reference repo')
+    args = parser.parse_args()
+
+    print(args)
+
+    work_path = args.work_path
+    if not os.path.exists(work_path):
+        os.mkdir(work_path)
+    cppcheck_path = os.path.join(work_path, 'cppcheck')
+
+    jobs = '-j' + str(args.j)
+    result_file = os.path.join(work_path, args.o)
+    my_root = os.path.dirname(os.path.dirname(os.path.abspath(sys.argv[0])))
+
+    if not lib.get_cppcheck(cppcheck_path, work_path):
```

Comment:
I think it would be good if you don't have to sync your repo with upstream master.

I suggest that we clone the repo and then checkout the revision that is in the users master.

Reply:
> I think it would be good if you don't have to sync your repo with upstream master.

I didn't get you. This code getting master to the some working_directory.
I assume that developer on his branch with changes running this script. Technically speaking it could be also not commited changes.

"Your" repo just building several lines below.

Reply:
The `get_cppcheck` contains this code:

    subprocess.check_call(['git', 'clone', 'https://github.com/danmar/cppcheck.git', cppcheck_path])

That will get the latest upstream commits even if those have not been pulled into your own repo yet. I don't want that the diffs caused by those commits are shown in the report.

Maybe you can lookup what revision the user 'master' is and checkout that..

    cd <users cppcheck folder> && git show --oneline -s master --> get revision
    cd <work_folder>/cppcheck && git checkout <revision>

Reply:
Perhaps `git merge-base` can be used instead? That would give the last common commit of master and branch. To avoid cases where someone has fetched but not updated, you'd to look at origin/master (and hope they didn't use a different name for the remote (this can most likely worked around)).

Reply:
ok I did not know about merge-base but I'll trust you on this.

Reply:
Ok, now it's clear for me. It's a really good point. Script updated